variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/factories/VariantVcfFactory.java

```diff
@@ -0,0 +1,355 @@
+
+import org.apache.commons.lang3.StringUtils;
+
+import uk.ac.ebi.eva.commons.core.models.VariantKeyFields;
+import uk.ac.ebi.eva.commons.core.models.genotype.Genotype;
+import uk.ac.ebi.eva.commons.core.models.pipeline.Variant;
+import uk.ac.ebi.eva.commons.core.models.pipeline.VariantSourceEntry;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+
+/**
+ * Class that parses VCF lines to create Variants.
+ */
+public class VariantVcfFactory {
+
+    /**
+     * Creates a list of Variant objects using the fields in a record of a VCF
+     * file. A new Variant object is created per allele, so several of them can
+     * be created from a single line.
+     * <p>
+     * Start/end coordinates assignment tries to work as similarly as possible
+     * as Ensembl does, except for insertions, where start is greater than end:
+     * http://www.ensembl.org/info/docs/tools/vep/vep_formats.html#vcf
+     *
+     * @param fileId,
+     * @param studyId
+     * @param line Contents of the line in the file
+     * @return The list of Variant objects that can be created using the fields from a VCF record
+     */
+    public List<Variant> create(String fileId, String studyId, String line) throws IllegalArgumentException {
+        String[] fields = line.split("\t");
+        if (fields.length < 8) {
+            throw new IllegalArgumentException("Not enough fields provided (min 8)");
+        }
+
+        String chromosome = getChromosomeWithoutPrefix(fields);
+        int position = getPosition(fields);
+        Set<String> ids = getIds(fields);
+        String reference = getReference(fields);
+        String[] alternateAlleles = getAlternateAlleles(fields, chromosome, position, reference);
+        float quality = getQuality(fields);
+        String filter = getFilter(fields);
+        String info = getInfo(fields);
+        String format = getFormat(fields);
+
+        List<VariantKeyFields> generatedKeyFields = buildVariantKeyFields(chromosome, position, reference,
+                alternateAlleles);
+
+        List<Variant> variants = new LinkedList<>();
+        // Now create all the Variant objects read from the VCF record
+        for (int altAlleleIdx = 0; altAlleleIdx < alternateAlleles.length; altAlleleIdx++) {
+            VariantKeyFields keyFields = generatedKeyFields.get(altAlleleIdx);
+            Variant variant = new Variant(chromosome, keyFields.getStart(), keyFields.getEnd(), keyFields.getReference(),
+                                          keyFields.getAlternate());
+            String[] secondaryAlternates = getSecondaryAlternates(altAlleleIdx, alternateAlleles);
+            VariantSourceEntry file = new VariantSourceEntry(fileId, studyId, secondaryAlternates, format, null, null,
+                                                             null);
```

Comment:
It would be preferable to add a new `VariantSourceEntry` constructor with only 4 arguments that delegates into the 7-argument one.